src/plugins/workspace/server/plugin.ts

```diff
@@ -134,7 +134,9 @@ export class WorkspacePlugin implements Plugin<WorkspacePluginSetup, WorkspacePl
           }
           const [coreStart] = await core.getStartServices();
           const uiSettingsClient = coreStart.uiSettings.asScopedToClient(
-            coreStart.savedObjects.getScopedClient(request)
+            coreStart.savedObjects.getScopedClient(request, {
+              excludedWrappers: [WORKSPACE_SAVED_OBJECTS_CLIENT_WRAPPER_ID],
+            })
           );
           const defaultWorkspaceId = await uiSettingsClient.get(DEFAULT_WORKSPACE);
```

Comment:
how about call `await uiSettingsClient.get(DEFAULT_WORKSPACE);` no matter what's the path after the user authenticated, after this call setting object should been initialized which include upgrade logic. With this, we don't need to put exclude workspace wrapper logic in other place.

Reply:
Get your point, yeah let me see if I can find a centralized place that will trigger the `get` method of ui settings client.